applications/RANSApplication/custom_utilities/rans_calculation_utilities.cpp

```diff
@@ -242,25 +325,124 @@ double CalculateLogarithmicYPlusLimit(
     return y_plus;
 }
 
-double GetDivergence(
-    const Geometry<ModelPart::NodeType>& rGeometry,
-    const Variable<array_1d<double, 3>>& rVariable,
-    const Matrix& rShapeDerivatives,
-    const int Step)
+double CalculateWallHeight(
+    const ConditionType& rCondition,
+    const array_1d<double, 3>& rNormal)
 {
-    double value = 0.0;
-    const int number_of_nodes = rGeometry.PointsNumber();
-    const int dim = rShapeDerivatives.size2();
+    KRATOS_TRY
 
-    for (int i = 0; i < number_of_nodes; ++i) {
-        const array_1d<double, 3>& r_value =
-            rGeometry[i].FastGetSolutionStepValue(rVariable, Step);
-        for (int j = 0; j < dim; ++j) {
-            value += r_value[j] * rShapeDerivatives(i, j);
+    array_1d<double, 3> normal = rNormal / norm_2(rNormal);
+
+    const ElementType& r_parent_element = rCondition.GetValue(NEIGHBOUR_ELEMENTS)[0];
+
+    const GeometryType& r_parent_geometry = r_parent_element.GetGeometry();
+    const GeometryType& r_condition_geometry = rCondition.GetGeometry();
+
+    auto calculate_cell_center = [](const GeometryType& rGeometry) -> array_1d<double, 3> {
+        const int number_of_nodes = rGeometry.PointsNumber();
+        array_1d<double, 3> cell_center = ZeroVector(3);
+        for (int i_node = 0; i_node < number_of_nodes; ++i_node) {
+            noalias(cell_center) =
+                cell_center + rGeometry[i_node].Coordinates() *
+                                  (1.0 / static_cast<double>(number_of_nodes));
+        }
+
+        return cell_center;
+    };
+
+    const array_1d<double, 3>& parent_center = calculate_cell_center(r_parent_geometry);
+
+    const array_1d<double, 3>& condition_center =
+        calculate_cell_center(r_condition_geometry);
+
+    return inner_prod(condition_center - parent_center, normal);
+
+    KRATOS_CATCH("");
+}
+
+array_1d<double, 3> CalculateWallVelocity(
+    const ConditionType& rCondition)
+{
+    array_1d<double, 3> normal = rCondition.GetValue(NORMAL);
+    normal /= norm_2(normal);
+
+    const ElementType& r_parent_element = rCondition.GetValue(NEIGHBOUR_ELEMENTS)[0];
+    const GeometryType& r_parent_geometry = r_parent_element.GetGeometry();
+
+    Vector parent_gauss_weights;
+    Matrix parent_shape_functions;
+    GeometryData::ShapeFunctionsGradientsType parent_shape_function_derivatives;
+    CalculateGeometryData(r_parent_geometry, GeometryData::IntegrationMethod::GI_GAUSS_1,
+                          parent_gauss_weights, parent_shape_functions,
+                          parent_shape_function_derivatives);
+
+    const Vector& gauss_parent_shape_functions = row(parent_shape_functions, 0);
+    const array_1d<double, 3>& parent_center_velocity =
+        EvaluateInPoint(r_parent_geometry, VELOCITY, gauss_parent_shape_functions);
+    const array_1d<double, 3>& parent_center_mesh_velocity = EvaluateInPoint(
+        r_parent_geometry, MESH_VELOCITY, gauss_parent_shape_functions);
```

Comment:
```suggestion
    const auto& parent_center_velocity =
        EvaluateInPoint(r_parent_geometry, VELOCITY, gauss_parent_shape_functions);
    const auto& parent_center_mesh_velocity = EvaluateInPoint(
        r_parent_geometry, MESH_VELOCITY, gauss_parent_shape_functions);
```
Same comment as above, this requires looping the nodes twice.

Reply:
done. Now using the Parameter Pack based EvaluateInPoint method :)